consensus/block_test.go

```diff
@@ -81,6 +82,52 @@ func TestConsensus(t *testing.T) {
 	}
 }
 
+func exchangeMessages(blocks []*Block, beEvil bool) {
+	firstEvil := false
+
+	for _, block := range blocks {
+		messages := block.OutgoingMessages()
+
+		for _, block2 := range blocks {
+			if block == block2 {
+				continue
+			}
+
+			for _, message := range messages {
+				if (beEvil && !firstEvil) {
+					switch m := message.(type) {
+					case *PrepareMessage:
+						m.Hn = math.MaxInt32
+						firstEvil = true
+					}
+				}
+
+				block2.Handle(block.publicKey, message)
+			}
+		}
+	}
+}
+
+func TestProtectionAgainstBigRangeDDoS(t *testing.T) {
```

Comment:
does this test infinite-ish-loop the code before your changes?

Reply:
Yep, hangs without the changes.